tango/common/logging.py

```diff
@@ -43,74 +178,387 @@ def critical_once(self, msg, *args, **kwargs):
-"""
-The log level.
-"""
+
+_LOGGING_HOST: str = os.environ.get("TANGO_LOGGING_HOST", "localhost")
+_LOGGING_PORT: Optional[int] = _parse_optional_int(os.environ.get("TANGO_LOGGING_PORT", None))
+_LOGGING_SERVER: Optional[LogRecordSocketReceiver] = None
+_LOGGING_SERVER_THREAD: Optional[threading.Thread] = None
+
+
+logging.setLoggerClass(TangoLogger)
 
 
 click_logger = logging.getLogger("click")
+"""
+A logger that logs messages through
+`click <https://click.palletsprojects.com/>`_'s
+``click.echo()`` function.
+
+This is provides a convenient way for command-line apps to log pretty, styled messages.
+"""
+
 click_logger.propagate = False
 
 
 class ClickLoggerHandler(logging.Handler):
     def emit(self, record: logging.LogRecord) -> None:
-        click.echo(record.getMessage())
+        if FILE_FRIENDLY_LOGGING:
+            click.echo(click.unstyle(record.getMessage()))
+        else:
+            click.echo(record.getMessage())
 
 
 click_logger.addHandler(ClickLoggerHandler())
-click_logger.disabled = (
-    True  # This is disabled by default, in case nobody calls initialize_logging().
-)
+click_logger.disabled = TANGO_CLICK_LOGGER_ENABLED
+
+
+def get_formatter() -> TangoFormatter:
+    log_format = "[%(process)d %(asctime)s %(levelname)s %(name)s] %(message)s"
+    return TangoFormatter(log_format)
 
 
 def initialize_logging(
     *,
     log_level: Optional[str] = None,
-    enable_click_logs: bool = False,
+    enable_click_logs: Optional[bool] = None,
+    file_friendly_logging: Optional[bool] = None,
+):
+    """
+    Initialize logging, which includes setting the global log level, format, and configuring
+    handlers.
+
+    .. tip::
+        This should be called as early on in your script as possible.
+
+    .. tip::
+        You should also call :func:`teardown_logging()` as the end of your script.
+
+    .. tip::
+        For worker threads/processes, use :func:`initialize_worker_logging()` instead.
+
+    Parameters
+    ----------
+    log_level : :class:`str`
+        Can be one of "debug", "info", "warning", "error". Defaults to the value
+        of :data:`TANGO_LOG_LEVEL`.
+    enable_click_logs : :class:`bool`
+        Set to ``True`` to enable messages from the :data:`click_logger`.
+    file_friendly_logging : :class:`bool`
+        Enable or disable file friendly logging. Defaults to the value of :data:`FILE_FRIENDLY_LOGGING`.
+
+    """
+    import multiprocessing as mp
+
+    is_main_process: bool
+    if hasattr(mp, "parent_process"):  # python 3.8 or greater
+        is_main_process = mp.parent_process() is None  # type: ignore
+    else:
+        is_main_process = mp.current_process().name == "MainProcess"
+
+    _initialize_logging(
+        log_level=log_level,
+        enable_click_logs=enable_click_logs,
+        file_friendly_logging=file_friendly_logging,
+        main_process=is_main_process,
+    )
+
+
+def initialize_worker_logging(worker_rank: Optional[int] = None):
```

Comment:
Since we can autodetect whether we're a worker or a main process, can't we just go down the correct code path automatically, so users only have to call `initialize_logging()` regardless of whether they are workers or main processes?

Reply:
:+1: